src/Datasource/RepositoryInterface.php

```diff
@@ -128,7 +132,7 @@ public function updateAll($fields, $conditions): int;
      * @return int Returns the number of affected rows.
      * @see \Cake\Datasource\RepositoryInterface::delete()
      */
-    public function deleteAll($conditions): int;
+    public function deleteAll(mixed $conditions): int;
```

Comment:
Could this be narrowed down?

Reply:
I don't know. There is a big abstraction between repository and tables. Not sure if other datasource specify conditions differently.

Reply:
Okay, thx.

Reply:
With `exists(array $conditions)`, wouldnt it be fair to also assume we could use similar limitations on this one?

Reply:
I think this type could be narrowed to `QueryExpression|Closure|array|string` like updateAll() is. The same for exists().

Reply:
Those are for fields. $conditions is still `mixed`. It says it accepts whatever `where()` does which is:

`where(Closure|array|string|null $conditions = null`

We could narrow down to that for those parameters.